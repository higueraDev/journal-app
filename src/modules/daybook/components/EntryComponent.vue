<template>
	<div
		@click="$router.push({ name: 'entry', params: { id: entry.id } })"
		class="entry-container mb-3 pointer p-2"
	>
		<div class="entry-title d-flex">
			<span class="text-success fs-5 fw-bold">{{day}}</span>
			<span class="mx-2 fs-5">{{month}}</span>
			<span class="mx-2 fw-light">{{yearDay}}</span>
		</div>
		<div class="entry-description text-start">
			{{ shortText }}
		</div>
	</div>
</template>

<script>
import getDayMonthYear from "@/modules/daybook/helpers/getDayMonthYear";
export default {
	props: {
		entry: {
			type: Object,
			required: true,
		},
	},
	computed: {
		shortText() {
			return this.entry.text.length > 130
				? this.entry.text.slice(0, 130) + "..."
				: this.entry.text;
		},
		day() {
			const { day } = getDayMonthYear(this.entry.date);
			return day;
		},
		month() {
			const { month } = getDayMonthYear(this.entry.date);
			return month;
		},
		yearDay() {
			const { yearDay } = getDayMonthYear(this.entry.date);
			return yearDay;
		},
	},
};
</script>

<style lang="scss" scoped>
.entry-container {
	border-bottom: 1px solid #2c3e50;
	transition: 0.2s all ease-in;

	&:hover {
		background-color: lighten($color: gray, $amount: 45);
	}
}

.entry-description {
	font-size: 12px;
}
</style>
