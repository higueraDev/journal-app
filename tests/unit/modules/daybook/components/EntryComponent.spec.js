import { shallowMount } from "@vue/test-utils";
import EntryComponent from "@/modules/daybook/components/EntryComponent.vue";
import { journalState } from "../../../mock-data/test-journal-state";
import getDayMonthYear from "@/modules/daybook/helpers/getDayMonthYear";

describe("Pruebas en el EntryComponent", () => {
	const mockRouter = {
		push: jest.fn(),
	};
	const [entry] = journalState.entries;
	const wrapper = shallowMount(EntryComponent, {
		props: {
			entry,
		},
		global: {
			mocks: {
				$router: mockRouter,
			},
		},
	});
	test("Debe hacer match con el snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});
	test("Debe redireccionar al hacer click en el entry-container", () => {
		const entryContainer = wrapper.find(".entry-container");
		entryContainer.trigger("click");
		expect(mockRouter.push).toBeCalled();
		expect(mockRouter.push).toBeCalledWith({
			name: "entry",
			params: { id: entry.id },
		});
	});
	test("Pruebas en las propiedades computadas", () => {
		const { day, month, yearDay } = getDayMonthYear(entry.date);

		expect(wrapper.vm.day).toBe(day);
		expect(wrapper.vm.month).toBe(month);
		expect(wrapper.vm.yearDay).toBe(yearDay);
	});
});
