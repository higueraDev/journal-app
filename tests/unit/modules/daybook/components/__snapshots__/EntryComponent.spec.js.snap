// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryComponent Debe hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">24</span><span class="mx-2 fs-5">Mayo</span><span class="mx-2 fw-light">2022,Martes</span></div>
  <div class="entry-description text-start">Fuimos por una nieve</div>
</div>
`;
